homeassistant/components/zwave_mqtt/binary_sensor.py

```diff
@@ -0,0 +1,400 @@
+        return self.values.primary.value
+
+    @property
+    def entity_registry_enabled_default(self) -> bool:
+        """Return if the entity should be enabled when first added to the entity registry."""
+        # Legacy binary sensors are phased out (replaced by notification sensors)
+        # Disable by default to not confuse users
+        for item in self.values.primary.node.values():
+            if item.command_class == CommandClass.NOTIFICATION:
+                # This device properly implements the Notification CC, legacy sensor can be disabled
+                return False
+        return True
+
+
+class ZWaveListValueSensor(ZWaveDeviceEntity, BinarySensorEntity):
+    """Representation of a binary_sensor from values in the Z-Wave Notification CommandClass."""
+
+    def __init__(
+        self,
+        values,
+        on_value,
+        device_class=None,
+        default_enabled=True,
+        off_value=NOTIFICATION_VALUE_CLEAR,
+    ):
+        """Initialize a ZWaveListValueSensor entity."""
+        super().__init__(values)
+        self._on_value = on_value
+        self._device_class = device_class
+        self._default_enabled = default_enabled
+        self._off_value = off_value
+        # make sure the correct value is selected at startup
+        self._state = False
+        self.on_value_update()
+
+    @callback
+    def on_value_update(self):
+        """Call when a value is added/updated in the underlying EntityValues Collection."""
+        if self.values.primary.value["Selected_id"] == self._on_value:
+            # Only when the active ID exactly matches our watched ON value, set sensor state to ON
+            self._state = True
+        elif self.values.primary.value["Selected_id"] == self._off_value:
+            # Only when the active ID exactly matches our watched OFF value, set sensor state to OFF
+            self._state = False
+        elif (
+            self._off_value is None
+            and self.values.primary.value["Selected_id"] != self._on_value
+        ):
+            # Off value not explicitly specified
+            # Some values are reset by the simple fact they're overruled by another value coming in
+            # For example the battery charging values in Power Management Index
+            self._state = False
+
+    @property
+    def name(self):
+        """Return the name of the entity."""
+        # Append value label to base name
+        base_name = super().name
+        value_label = ""
+        for item in self.values.primary.value["List"]:
+            if item["Value"] == self._on_value:
+                value_label = item["Label"]
+                break
+        # Strip "on location" / "at location" from name
+        # Note: We're assuming that we don't retrieve 2 values with different location
+        value_label = value_label.split(" on ")[0]
+        value_label = value_label.split(" at ")[0]
+        return f"{base_name} - {value_label}"
```

Comment:
I think we use colon in the base class. Maybe stick with that?
```suggestion
        return f"{base_name}: {value_label}"
```

Reply:
Nope, this is something else ;-)
This will result in these labels:

Motion Sensor (PIR): Home Security - Motion Detected

Reply:
Why can't we continue with the colons? This is how my light is named by default: `LED Bulb 6:Multi-Colour: Level`.

Reply:
I'm thinking that it will look more consistent with just one type of divider. But I'll leave it up to you to decide.

Reply:
Sure, no problem. I do not have a preference.